src/cascadia/WindowsTerminal/WindowEmperor.cpp

```diff
@@ -0,0 +1,690 @@
+        wil::unique_any<LPWSTR*, decltype(&::LocalFree), ::LocalFree> argv{ CommandLineToArgvW(commandline, &argc) };
+        if (argv)
+        {
+            for (auto& elem : wil::make_range(argv.get(), argc))
+            {
+                args.emplace_back(elem);
+            }
+        }
+    }
+    if (args.empty())
+    {
+        args.emplace_back(L"wt.exe");
+    }
+}
+
+bool WindowEmperor::HandleCommandlineArgs()
+{
+    std::vector<winrt::hstring> args;
+    _buildArgsFromCommandline(args);
+    auto cwd{ wil::GetCurrentDirectoryW<std::wstring>() };
+
+    Remoting::CommandlineArgs eventArgs{ { args }, { cwd } };
+
+    const auto isolatedMode{ _app.Logic().IsolatedMode() };
+
+    const auto result = _manager.ProposeCommandline(eventArgs, isolatedMode);
+
+    if (result.ShouldCreateWindow())
+    {
+        CreateNewWindowThread(Remoting::WindowRequestedArgs{ result, eventArgs }, true);
+
+        _manager.RequestNewWindow([this](auto&&, const Remoting::WindowRequestedArgs& args) {
+            CreateNewWindowThread(args, false);
+        });
+
+        _becomeMonarch();
+    }
+    else
+    {
+        const auto res = _app.Logic().GetParseCommandlineMessage(eventArgs.Commandline());
+        if (!res.Message.empty())
+        {
+            AppHost::s_DisplayMessageBox(res);
+            ExitThread(res.ExitCode);
+        }
+    }
+
+    return result.ShouldCreateWindow();
+}
+
+void WindowEmperor::WaitForWindows()
+{
+    MSG message;
+    while (GetMessage(&message, nullptr, 0, 0))
+    {
+        TranslateMessage(&message);
+        DispatchMessage(&message);
+    }
+}
+
+void WindowEmperor::CreateNewWindowThread(Remoting::WindowRequestedArgs args, const bool /*firstWindow*/)
+{
+    Remoting::Peasant peasant{ _manager.CreatePeasant(args) };
+
+    auto window{ std::make_shared<WindowThread>(_app.Logic(), args, _manager, peasant) };
+
+    window->Started([this, sender = window]() -> winrt::fire_and_forget {
+        // Add a callback to the window's logic to let us know when the window's
+        // quake mode state changes. We'll use this to check if we need to add
+        // or remove the notification icon.
+        sender->Logic().IsQuakeWindowChanged([this](auto&&, auto &&) -> winrt::fire_and_forget {
+            co_await wil::resume_foreground(this->_dispatcher);
+            this->_checkWindowsForNotificationIcon();
+        });
+        sender->UpdateSettingsRequested([this]() -> winrt::fire_and_forget {
+            // We MUST be on the main thread to update the settings. We will crash when trying to enumerate fragment extensions otherwise.
+            co_await wil::resume_foreground(this->_dispatcher);
+            _app.Logic().ReloadSettings();
+        });
+
+        // These come in on the sender's thread. Move back to our thread.
+        co_await wil::resume_foreground(_dispatcher);
+
+        _windows.push_back(std::move(sender));

```

Comment:
Captured variables of a lambda don't survive a `co_await`: https://devblogs.microsoft.com/oldnewthing/20211103-00/?p=105870
Additionally, since the lambda is not declared as `mutable`, all its members are `const`, which means that the `std::move` is ineffective. (Dustin would probably be happy to know that clang-tidy can warn about this. 😄)